### ISSUE TITLE
Match Differencify and Puppeteer API 

### ISSUE THREAD

**Issue by NimaSoroush** [NimaSoroush/differencify#53]
**Context:**
From [ms88privat](https://github.com/NimaSoroush/differencify/issues/51#issuecomment-344277420)'s comment: 
- It would be hard for users to learn Differencify's API on top of Puppeteer API
- It would be hard to keep Differencify's API up2date with Puppeteer's API

**Goal:**
- Forward all method calls that are not matching Differencify's API interface directly to Puppeteer.
- Differencify should be still able to override Puppeteer's API if it is necessary.
- This should not introduce any breaking changes to difference's interface

CC: @ms88privat @badsyntax

**Comment by NimaSoroush** [comment]
I have spent some time last night to make a [WIP](https://github.com/NimaSoroush/differencify/pull/54) of this task. The concept works fine. Just need to fix/write additional test around it

**Comment by ms88privat** [comment]
Wow, that was fast 👍 . I don't understand this code by now completely. For now we can only chain methods on the page object, right? How about clicking multiple elements on the page before taking the screenshot? Maybe this chaining stuff gets to complex and I don't see as much benefit when you can just use async await today. I'm on the wrong path or what do you think about that?

**Comment by NimaSoroush** [comment]
@ms88privat : Yeah, currently we only chain over page object as most of the puppeteer logic is around that area, but it is doable to it over browser object as well, just needs a bit of refactoring.

I don't see any problem with having a long chain of steps. The longest chained scenario would still be shorter than separate async/await. But still if someone really doesn't like the concept of chaining steps there is `{ chain: false }` available
```
const page = await differencify.init({ chain: false });
await page.goto('https://github.com/NimaSoroush/differencify');
...
```

**Comment by ms88privat** [comment]
But what about using `page.$()`  and then continue with the`ElementHandle` and back to the `page` again? I think most of the tests inside a SPA will need this, or this is what I'm imaging to using this tool for.

**Comment by NimaSoroush** [comment]
That would be possible as well with `result()` function which exists (but not documented 😄 ) [here](https://github.com/NimaSoroush/differencify/blob/bad4af39126c162651743fa15c7b0c59ae153620/src/helpers/proxyChain.js#L76)

I am thinking about this syntax as the end result:
```
    await differencify
      .launchBrowser()
      .newPage()
      .tracing.
        start({path: 'trace.json'})
      .resize({ width: 1600, height: 1200 })
      .goto('https://github.com/NimaSoroush/differencify')
      .on('console', msg => console.log(msg.text))
      .$('input[type=submit]')
        .result(inputElement => {
          inputElement.click();
        })
      .wait(1000)
      .keyboard
        .press('Enter')
        .type('Hello')
      .mouse
        .click(120, 240)
      .tracing.
        start({path: 'trace.json'})
      .capture()
      .toMatchSnapshot()
      .close()
      .end();
```
Thought?
CC: @ms88privat @badsyntax @danrr

**Comment by ms88privat** [comment]
lgtm! Will look more closely into the source code and building stuff with it next week.

**Comment by badsyntax** [comment]
The API is nice. So is `newPage` going to return a differencify `Page` instance which has some methods like `resize` and `toMatchSnapshot` etc but all other calls will be proxied to browser page API?

**Comment by NimaSoroush** [comment]
@badsyntax : Most of the functionality is around `page` object, but yeah none-`page` calls will be proxied to browser

**Comment by NimaSoroush** [comment]
Closed by [#54]
